addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/handler/ExecHandler.java

```diff
@@ -166,7 +174,34 @@ public void run() {
                     logger.error("An exception occurred while executing '{}' : '{}'",
                             new Object[] { commandLine.toString(), e.getMessage() });
                     updateState(RUN, OnOffType.OFF);
-                    updateState(OUTPUT, new StringType(e.getMessage()));
+
+                    if (e.getMessage().equals("OPEN")) {
+                        updateState(CONTACT_OUTPUT, OpenClosedType.OPEN);
+                    }
+
+                    if (e.getMessage().equals("CLOSED")) {
+                        updateState(CONTACT_OUTPUT, OpenClosedType.CLOSED);
+                    }
+
+                    if (e.getMessage().equals("ON")) {
+                        updateState(DIMMER_OUTPUT, OnOffType.ON);
+                        updateState(SWITCH_OUTPUT, OnOffType.ON);
+                    }
+
+                    if (e.getMessage().equals("OFF")) {
+                        updateState(DIMMER_OUTPUT, OnOffType.OFF);
+                        updateState(SWITCH_OUTPUT, OnOffType.OFF);
+                    }
+
+                    if (e.getMessage().equals("UP")) {
+                        updateState(ROLLERSHUTTER_OUTPUT, UpDownType.UP);
+                    }
+
+                    if (e.getMessage().equals("DOWN")) {
+                        updateState(ROLLERSHUTTER_OUTPUT, UpDownType.DOWN);
+                    }
+
+                    updateState(STRING_OUTPUT, new StringType(e.getMessage()));
```

Comment:
Yes. Looking at it I just noticed I put this code at the wrong place. 

It is not missing, simply said, if the external command execution results in an output "ON" that is  translated to an ON command. This is handy for scripts that output such a String. 
Anything else is just set to the "string" output channel.